### PR TITLE
Enforce max Java bytecode version in Maven dependencies with `enforceBytecodeVersion` rule 

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- Enforce max Java bytecode version in dependencies with `enforceBytecodeVersion` rule from MojoHaus‘s Extra Enforcer Rules (https://www.mojohaus.org/extra-enforcer-rules/enforceBytecodeVersion.html)
- Prevent accidentally bumping or introducing maven dependencies a Java version over baseline (currently Java 8)

Currently, some dependencies remain on older versions for Java 8 compatibility. This enforceBytecodeVersion  rule will help to enforce them aligning to Java 8. 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
For example, the lastest release version of Hikari is 5.0.1 required Java 11. If we bump it to 5.0.1, the enforcer will check `enforceBytecodeVersion` rule and fails with the error message.

```
[INFO] Restricted to JDK 8 yet com.zaxxer:HikariCP:jar:5.0.1:compile contains com/zaxxer/hikari/HikariConfig.class targeted to JDK 11
[ERROR] Rule 1: org.apache.maven.plugins.enforcer.EnforceBytecodeVersion failed with message:
Found Banned Dependency: com.zaxxer:HikariCP:jar:5.0.1
```

File: pom.xml [hunk 1]
```diff
@@ -212,6 +212,7 @@
 
         <maven.plugin.build.helper.version>3.2.0</maven.plugin.build.helper.version>
         <maven.plugin.download.version>1.6.6</maven.plugin.download.version>
+        <maven.plugin.enforcer.mojo.extra.rules.version>1.6.1</maven.plugin.enforcer.mojo.extra.rules.version>
```
Comment (pan3793) [inline pom.xml:215]: maven.plugin.enforcer.extra.rules.version

**Review by pan3793 — APPROVED** [review]
LGTM, except to the property name

File: pom.xml (hunk 1 above)
Comment (bowenliang123) [reply pom.xml:215]: `mojo` should be kept in the prop name, as `org.codehaus.mojo:extra-enforcer-rules` is third-party rules provider for mavn enforcer plugin.

Comment (pan3793) [reply pom.xml:215]: Under the premise of being identifiable, we prefer the short name.

Comment (bowenliang123) [reply pom.xml:215]: What about `maven.plugin.enforcer.mojo.rules.version` ?

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**